groupStatusesWriter.php

```diff
@@ -1,119 +1,75 @@
 <?php
 
-    function get_values(){
-      $params = parse_ini_file('config');
-      return array_values($params);
+class JenkinsConfig
+{
+    public function get_values(){
+      return array_values(parse_ini_file('config'));
     }
 
-    function get_user(){
-        return get_values()[0];
+    public function get_user(){
+        return $this->get_values()[0];
     }
 
-    function get_pass(){
-        return get_values()[1];
+    public function get_pass(){
+        return $this->get_values()[1];
     }
 
-    function get_site(){
-        return get_values()[2];
+    public function get_site(){
+        return $this->get_values()[2];
     }
 
-    function get_url(){
-      return "https://".get_user().":".get_pass()."@".get_site()."/api/xml";
+    public function get_url(){
+      return "https://".$this->get_user().":".$this->get_pass()."@".$this->get_site()."/api/xml";
     }
 
-    function prior_check($jobName) {
-        if(file_exists('results/fails.xml')){
-            $pastFails = simplexml_load_file('results/fails.xml');
-        } else {
-            touch('results/fails.xml');
-            chmod('results/fails.xml', 0777);
-            $pastFails = simplexml_load_file('results/fails.xml');
-        }
-        
-        if($pastFails){
-          foreach ($pastFails as $results) {
-            if($jobName == $results->name) {
-              return true;
+    public function writeToStatusXml() {
+        if($xml = file_get_contents($this->get_url())){
+            if(!file_exists('results/status.xml')){
+                mkdir('results', 0664, true);
             }
-          }
+            file_put_contents('results/status.xml' , $xml);
+        } else {
+            exit('Could not get xml');
         }
-        return false;
     }
 
-    function get_culprit($jobName) {      
-      if($xml = simplexml_load_file('https://'.get_user().':'.get_pass().'@'.get_site().'/job/'.$jobName.'/lastFailedBuild/api/xml')) {
+    public function getAllJobsPath() {
+        return 'json/allJobs.json';
+    }
+}
+
+class Culprit
+{
+    function get_culprit($jobName) { 
+      $con = new JenkinsConfig;             
+      if($xml = simplexml_load_file('https://'.$con->get_user().':'.$con->get_pass().'@'.$con->get_site().'/job/'.$jobName.'/lastFailedBuild/api/xml')) {
           if(isset($xml->culprit->fullName)){
             $culprit = $xml->culprit->fullName;
           } else {
-            $culprit = "No culprit";
+            $culprit = "No culprit returned";
           }
       } else {
-          $culprit = $jobName." appears to be in good shape.";
+          $culprit = "No last fail";
```

Comment:
how about 'No failed build returned'? to me this sounds better, but open for debate